challenges/functions.js

```diff
@@ -6,25 +6,39 @@
   * The last parameter accepts a callback 
   * In the body of the function return the callback with the two parameters that you created
 */
-
+function consume(arg1, arg2, cb){
+  return cb(arg1, arg2);
+}
 
 /* Step 2: Create several functions to callback with consume();
   * Create a function named add that returns the sum of two numbers
   * Create a function named multiply that returns the product of two numbers 
   * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
 */
+function add(x, y){
+  return x + y;
+}
 
+function multiply(x, y){
+  return x * y;
+}
+
+function greeting(first, last){
+  return `Hello ${first} ${last}, nice to meet you!`;
+}
 
 /* Step 3: Check your work by uncommenting the following calls to consume(): */
-//consume(2,2,add); // 4
-// consume(10,16,multiply); // 160
-// consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
+console.log(consume(2,2,add)); // 4
+console.log(consume(10,16,multiply)); // 160
+console.log(consume("Mary","Poppins", greeting)); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+/* Explanation: Scope! Imagine each function as a box, and each function inside of it as a smaller box stacked on top of it that can see everything below and around it.
+This explains why nestedFunction can access the variable internal. The nestedFunction box rests on top of the myFunction box and can see everything it has to offer. ;) 
+In other words, nested functions can access every variable that precedes them, but not the other way around.*/
```

Comment:
Exactly what I was looking for. When you're asked to write examples, try to imagine you're explaining it to a 10 year old. If you can explain it to a 10 year old, you can understand the work.